purchases/src/test/java/com/revenuecat/purchases/PurchasesTest.kt

```diff
@@ -449,28 +449,75 @@ class PurchasesTest {
     }
 
     @Test
-    fun `when making a deferred upgrade, completion is called with null purchase`() {
```

Comment:
We had 4 separate but very similar tests for DEFERRED mode, which I'm not entirely sure were even working properly (e.g. ProrationMode.DEFERRED was not even passed).

I simplified them into two tests, each using the old and the new purchase methods respectively.